app/services/induction/set_school_cohort_appropriate_body.rb

```diff
@@ -3,34 +3,30 @@
 class Induction::SetSchoolCohortAppropriateBody < BaseService
   def call
     ActiveRecord::Base.transaction do
-      set_school_cohort_appropriate_body
-      set_induction_records_appropriate_body if update_induction_records
+      id = appropriate_body_id unless appropriate_body_appointed == false
+      unknown = appropriate_body_appointed == false
+      set_induction_records_matching_school_ab(appropriate_body_id: id)
+      school_cohort.update!(appropriate_body_unknown: unknown, appropriate_body_id: id)
     end
   end
 
 private
 
-  attr_reader :school_cohort, :appropriate_body_id, :appropriate_body_appointed, :update_induction_records
+  attr_reader :school_cohort, :appropriate_body_id, :appropriate_body_appointed
 
-  def initialize(school_cohort:, appropriate_body_id:, appropriate_body_appointed:, update_induction_records: false)
+  def initialize(school_cohort:, appropriate_body_id:, appropriate_body_appointed:)
     @school_cohort = school_cohort
     @appropriate_body_id = appropriate_body_id
     @appropriate_body_appointed = appropriate_body_appointed
-    @update_induction_records = update_induction_records
   end
 
-  def set_school_cohort_appropriate_body
-    if appropriate_body_appointed == false
-      school_cohort.appropriate_body_unknown = true
-      school_cohort.appropriate_body = nil
-    else
-      school_cohort.appropriate_body_unknown = false
-      school_cohort.appropriate_body_id = appropriate_body_id
+  def set_induction_records_matching_school_ab(appropriate_body_id:)
+    school_cohort.induction_programmes
+                 .map(&:participant_profiles).flatten.uniq
+                 .map(&:latest_induction_record)
+                 .select(&:matches_school_appropriate_body?)
+                 .each do |induction_record|
+      induction_record.update!(appropriate_body_id:)
```

Comment:
Just a question about this, this looks like it will update the appropriate body on a participants latest induction record even if that record is now withdrawn or at a different school ie. they left this school and the new school happens to have the same AB?

Reply:
`induction_programme.participant_profiles` only pick those participants with active induction records.

Reply:
Ah so it does 👍